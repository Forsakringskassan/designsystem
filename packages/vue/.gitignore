# vite dev
/src/vite-dev/app.vue
