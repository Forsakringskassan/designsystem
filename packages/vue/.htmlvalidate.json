{
    "extends": ["@fkui/vue:recommended"],
    "plugins": ["./htmlvalidate"],
    "transform": {
        "^.*\\.spec.ts$": "html-validate-vue:html"
    }
}
