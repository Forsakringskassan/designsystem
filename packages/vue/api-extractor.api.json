{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "mainEntryPointFilePath": "<projectFolder>/temp/types/api.d.ts",
    "dtsRollup": {
        "enabled": false
    },
    "messages": {
        "compilerMessageReporting": {
            "default": {
                "logLevel": "none"
            }
        }
    }
}
