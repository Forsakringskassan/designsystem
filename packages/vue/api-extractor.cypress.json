{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "mainEntryPointFilePath": "<projectFolder>/temp/types/cypress/index.d.ts",
    "compiler": {
        "tsconfigFilePath": "<projectFolder>/tsconfig.pageobjects.json"
    },
    "apiReport": {
        "reportFileName": "<unscopedPackageName>-cypress.api.md"
    },
    "docModel": {
        "enabled": true,
        "apiJsonFilePath": "<projectFolder>/temp/cypress.api.json"
    },
    "dtsRollup": {
        "publicTrimmedFilePath": "<projectFolder>/dist/types/cypress.d.ts"
    },
    "messages": {
        "extractorMessageReporting": {
            "ae-unresolved-link": {
                "logLevel": "none"
            }
        }
    }
}
