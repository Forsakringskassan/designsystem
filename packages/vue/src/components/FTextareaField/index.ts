export { default as FTextareaField } from "./FTextareaField.vue";
