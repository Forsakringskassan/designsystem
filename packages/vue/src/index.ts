/* include typescript declarations for vite static asset handling (e.g. `?raw`) */
/// <reference types="vite/client" />

export * from "./components";
export {
    type UseSlotUtils,
    type WithoutInstance,
    useSlotUtils,
} from "./composables";
export {
    config,
    type FKUIConfig,
    FKUIConfigButtonOrder,
    type FKUIContext,
    type MaybeWithFKUIContext,
    setRunningContext,
} from "./config";
export * from "./internal-components";
export * from "./plugins";
export {
    type ComponentUnmountEvent,
    type ComponentValidityEvent,
    type ErrorItem,
    type GroupValidityEvent,
    type ListArray,
    type ListItem,
    type UnknownItem,
    ErrorData,
    ErrorViewData,
    FormErrorList,
    MenuAction,
} from "./types";
export * from "./utils";
