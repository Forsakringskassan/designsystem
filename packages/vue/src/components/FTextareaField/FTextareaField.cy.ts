import { type DefineComponent, defineComponent } from "vue";
import {
    DensityWrapper,
    densityWrapperHeight,
    densityWrapperWidth,
} from "@fkui/test-utils/vue";
import { FTextareaFieldPageObject } from "../../cypress";
import { FTooltip } from "../FTooltip";
import FTextareaField from "./FTextareaField.vue";

function createComponent(template: string): DefineComponent {
    return defineComponent({
        template,
        components: {
            FTextareaField,
            FTooltip,
        },
        data() {
            return {
                about: "",
            };
        },
    });
}

describe("FTextareaField", () => {
    it("should provide a page object that can access any necessary elements", () => {
        const template = /* HTML */ `
            <f-textarea-field
                id="berattelse-tooltip"
                v-model="about"
                v-test="'berattelse-text-field'"
                :maxlength="100"
                :soft-limit="20"
                characters-left-warning="Antal tecken kvar: %charactersLeft%"
            >
                <template #default> Berätta om dig själv </template>
                <template #tooltip>
                    <f-tooltip
                        screen-reader-text="Läs mer om berätta om dig själv"
                        header-tag="h1"
                    >
                        <template #header>
                            Mer om berätta om dig själv
                        </template>
                        <template #body>
                            Denna berättelsen kräver lite närmare förklaring.
                        </template>
                    </f-tooltip>
                </template>
                <template
                    #description="{ descriptionClass, formatDescriptionClass }"
                >
                    <span :class="descriptionClass">
                        En inte allt för utförlig berättelse
                    </span>
                    <span :class="formatDescriptionClass">
                        (max 100 tecken)
                    </span>
                </template>
            </f-textarea-field>
        `;
        cy.mount(createComponent(template));
        const berattelseTextareaField = new FTextareaFieldPageObject(
            '[data-test="berattelse-text-field"]',
        );
        berattelseTextareaField.label
            .el()
            .should("contain.text", "Berätta om dig själv");
        berattelseTextareaField.label
            .description()
            .should("contain.text", "En inte allt för utförlig berättelse");
        berattelseTextareaField.label
            .formatDescription()
            .should("contain.text", "(max 100 tecken)");
        berattelseTextareaField.label.errorMessage().should("not.exist");
        berattelseTextareaField.label.errorIcon().should("not.exist");

        berattelseTextareaField
            .input()
            .focus()
            .clear()
            .type("asd000")
            .should("have.value", "asd000")
            .should("have.focus")
            .blur()
            .should("not.have.focus");

        berattelseTextareaField.enter("qwe456");
        berattelseTextareaField.value().should("be.equal", "qwe456");

        berattelseTextareaField.tooltip.iButton().should("be.visible");
        berattelseTextareaField.tooltip.iButton().click();
    });

    /* eslint-disable-next-line mocha/no-skipped-tests -- temporary to get builds running */
    describe.skip("density", () => {
        const DensityComponent = defineComponent({
            template: /* HTML */ `
                <density-wrapper>
                    <f-textarea-field v-model="textAreaField" maxlength="100">
                        Flerradigt inmatningsfält
                    </f-textarea-field>
                </density-wrapper>
            `,
            components: {
                DensityWrapper,
                FTextareaField,
            },
            data() {
                return {
                    textAreaField: [1, 2, 3, 4]
                        .map((it) => `Rad ${it}`)
                        .join("\n"),
                };
            },
        });

        it(`should be densified`, () => {
            cy.viewport(densityWrapperWidth, densityWrapperHeight);
            cy.mount(DensityComponent);
            cy.toMatchScreenshot();
        });
    });
});
