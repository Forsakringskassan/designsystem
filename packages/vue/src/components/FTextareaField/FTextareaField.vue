<script lang="ts">
import { defineComponent } from "vue";
import { type ValidityEvent, isSet, ElementIdService } from "@fkui/logic";
import { FLabel } from "../FLabel";
import { renderSlotText, dispatchComponentValidityEvent } from "../../utils";

export default defineComponent({
    name: "FTextareaField",
    components: {
        FLabel,
    },
    inheritAttrs: false,
    props: {
        /**
         * The id for the input id attribute.
         * The id for the label for attribute.
         * If the prop is not set a random value will be generated.
         */
        id: {
            type: String,
            required: false,
            default: () => ElementIdService.generateElementId(),
        },
        /**
         * The value for the input.
         * If the prop is not set undefined will be used.
         * @model
         */
        modelValue: {
            type: String,
            required: false,
            default: undefined,
        },
        /**
         * The number of characters for when the "characters left" warning should be shown.
         * A value of 100 means that when 100 or less characters is left the warning is shown.
         * If softLimit is used, then maxlength is required.
         * If the prop is not set undefined will be used, which means that no warning will be shown.
         */
        softLimit: {
            type: Number,
            required: false,
            default: undefined,
        },
        /**
         * The maximum amount of characters permitted in the textarea.
         * If the prop is not set undefined will be used, which means unlimited.
         */
        maxlength: {
            type: Number,
            required: false,
            default: undefined,
        },
        /**
         * The string that should be shown in the "characters left" warning.
         * Must contain the word %charactersLeft% which is used to interpolate the number of characters left into the warning string.
         * If the prop is not set "Antal tecken kvar: %charactersLeft%" will be used.
         */
        charactersLeftWarning: {
            type: String,
            required: false,
            default: "Antal tecken kvar: %charactersLeft%",
        },
        /**
         * Specifies that the component should be disabled, i.e. unusable.
         */
        disabled: {
            type: Boolean,
            default: false,
        },
        /**
         * Enabling vertical resizing of the textarea
         */
        resizable: {
            type: Boolean,
            default: false,
        },
    },
    emits: ["input", "update:modelValue"],
    data() {
        return {
            validityMode: "INITIAL" as string,
            validationMessage: "" as string,
        };
    },
    computed: {
        attrs(): Record<string, unknown> {
            return {
                rows: 4,
                ...this.$attrs,
                value: this.modelValue,
                maxlength: this.maxlength,
            };
        },
        isValid(): boolean {
            return this.validityMode === "VALID";
        },
        hasError(): boolean {
            return this.validityMode === "ERROR";
        },
        validityClass(): Record<string, boolean> {
            return {
                ["textarea-field--error"]: this.hasError,
            };
        },
        charactersLeft(): number {
            if (this.modelValue) {
                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- required when `softLimit`
                return this.maxlength! - this.modelValue.length;
            } else {
                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- required when `softLimit`
                return this.maxlength!;
            }
        },
        showCharactersLeftWarning(): boolean {
            return (
                isSet(this.softLimit) &&
                isSet(this.modelValue) &&
                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- required when `softLimit`
                this.modelValue.length >= this.maxlength! - this.softLimit
            );
        },
        charactersLeftWarningInterpolated(): string {
            return `${this.charactersLeftWarning.replace("%charactersLeft%", this.charactersLeft.toString())}`;
        },
        textareaClass(): string[] {
            const classes = ["textarea-field__textarea"];

            if (this.resizable) {
                classes.push("textarea-field__resize--vertical");
            } else {
                classes.push("textarea-field__resize--none");
            }

            return classes;
        },
    },
    mounted() {
        if (isSet(this.softLimit) && !isSet(this.maxlength)) {
            throw new Error("You must pass a maxlength");
        }
    },
    methods: {
        onInput(event: Event): void {
            if (event.target instanceof HTMLTextAreaElement) {
                /**
                 * V-model event.
                 * @event update:modelValue
                 * @type {string}
                 */
                this.$emit("update:modelValue", event.target.value);

                /**
                 * @event input
                 * @type {string}
                 */
                this.$emit("input", event.target.value);
            }
        },
        onValidity({ detail }: CustomEvent<ValidityEvent>): void {
            this.validationMessage = detail.validationMessage;
            this.validityMode = detail.validityMode;
            const errorMessage = renderSlotText(this.$slots.default) ?? "";
            const element = this.$el.querySelector(`#${detail.elementId}`);
            if (element) {
                dispatchComponentValidityEvent(element, {
                    ...detail,
                    errorMessage,
                    focusElementId: detail.elementId,
                });
            }
        },
        onPendingValidity(): void {
            this.validityMode = "INITIAL";
        },
    },
});
</script>

<template>
    <div class="textarea-field" :class="validityClass">
        <f-label :for="id">
            <template #default>
                <!-- @slot Slot for label content. -->
                <slot name="default"></slot>
            </template>

            <template v-if="$slots.tooltip" #tooltip>
                <!-- @slot Slot for tooltip. -->
                <slot name="tooltip"></slot>
            </template>

            <template #description="{ descriptionClass, formatDescriptionClass }">
                <!--
                @slot Optional slot for description. See {@link FLabel} for details.
                @binding {string[]} description-class CSS classes for primary description content.
                @binding {string[]} format-description-class CSS classes for format description.
            -->
                <slot name="description" :description-class :format-description-class></slot>
            </template>

            <template #error-message>
                <!--
                @slot Slot for displaying single or several error messages.
                @binding {boolean} hasError Set to true when a validation error is present
                @binding {string} validationMessage Descriptive validation error message for current error
            -->
                <slot name="error-message" v-bind="{ hasError, validationMessage }">
                    <template v-if="hasError">{{ validationMessage }}</template>
                </slot>
            </template>
        </f-label>

        <f-label v-if="softLimit" :for="id" aria-live="polite">
            <template #description="{ descriptionClass }">
                <span v-if="showCharactersLeftWarning" :class="descriptionClass">
                    {{ charactersLeftWarningInterpolated }}
                </span>
            </template>
        </f-label>
        <textarea
            :id="id"
            :class="textareaClass"
            v-bind="attrs"
            :disabled="disabled"
            @input="onInput"
            @validity="onValidity"
            @pending-validity="onPendingValidity"
        ></textarea>
    </div>
</template>
