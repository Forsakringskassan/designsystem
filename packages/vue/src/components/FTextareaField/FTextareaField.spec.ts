import "html-validate/jest";
import { VueWrapper, mount } from "@vue/test-utils";
import {
    type ValidityEvent,
    type ValidatableHTMLElement,
    type PendingValidityEvent,
} from "@fkui/logic";
import { createPlaceholderInDocument } from "@fkui/test-utils/vue";
import flushPromises from "flush-promises";
import FTextareaField from "./FTextareaField.vue";

function createWrapper({ props = {}, slots = {}, attrs = {} } = {}): VueWrapper<
    InstanceType<typeof FTextareaField>
> {
    return mount(FTextareaField, {
        attrs: { ...attrs, id: "textarea-field" },
        props: { ...props },
        slots: { default: "Label", ...slots },
        global: {
            stubs: ["f-icon"],
        },
        attachTo: createPlaceholderInDocument(),
    });
}

describe("snapshots", () => {
    it("should match snapshot with label and textarea", () => {
        const wrapper = createWrapper();

        expect(wrapper.element).toMatchSnapshot();
    });

    it("should match snapshot with label, error message and textarea", () => {
        const wrapper = createWrapper({
            slots: { "error-message": "ERRROR_MESSAGE" },
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it("should match snapshot with label, tooltip, description, error message and textarea", () => {
        const wrapper = createWrapper({
            slots: {
                description: "DESCRIPTION",
                tooltip: "TOOLTIP",
                "error-message": "ERROR_MESSAGE",
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it("should match snapshot with resize vertical", () => {
        const wrapper = createWrapper({
            props: {
                resizable: true,
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it.each`
        validityMode | isValid
        ${"VALID"}   | ${true}
        ${"ERROR"}   | ${false}
        ${"INITIAL"} | ${true}
        ${"INITIAL"} | ${false}
    `(
        "should match snapshot when validityMode is $validityMode and isValid is $isValid",
        async ({ validityMode, isValid }) => {
            const wrapper = createWrapper({
                attrs: { id: "elementId" },
            });

            const textareaWrapper = wrapper.get("textarea");
            const htmlTextareaElement =
                textareaWrapper.element as HTMLTextAreaElement;

            htmlTextareaElement.dispatchEvent(
                new CustomEvent<ValidityEvent>("validity", {
                    detail: {
                        target: wrapper.element as ValidatableHTMLElement,
                        elementId: "elementId",
                        isValid,
                        validityMode,
                        validationMessage: "Something went wrong.",
                        nativeEvent: "input",
                    },
                }),
            );
            await flushPromises();
            wrapper.vm.$forceUpdate();

            expect(wrapper.element).toMatchSnapshot();
        },
    );
});

describe("attributes", () => {
    it("should pass attributes", () => {
        const wrapper = createWrapper({
            attrs: {
                required: true,
                rows: 8,
                maxlength: 20,
            },
            props: {
                disabled: true,
            },
        });
        const textareaWrapper = wrapper.get("textarea");

        expect(textareaWrapper.attributes("disabled")).toBeDefined();
        expect(textareaWrapper.attributes("required")).toBeDefined();
        expect(textareaWrapper.attributes("rows")).toBe("8");
        expect(textareaWrapper.attributes("maxlength")).toBe("20");
    });
});

describe("events", () => {
    it("should support v-model by emitting input event with value", () => {
        const wrapper = createWrapper({
            props: { modelValue: "Bana" },
        });
        const textareaWrapper = wrapper.get("textarea");
        const htmlTextAreaElement =
            textareaWrapper.element as HTMLTextAreaElement;

        expect(htmlTextAreaElement.value).toBe("Bana");
        textareaWrapper.setValue("Banana");
        expect(htmlTextAreaElement.value).toBe("Banana");
        expect(wrapper.emitted("input")![0][0]).toMatchInlineSnapshot(
            `"Banana"`,
        );
    });

    it("should pass listeners", async () => {
        const focus = jest.fn();
        const blur = jest.fn();

        const wrapper = createWrapper({
            attrs: {
                onFocus: focus,
                onBlur: blur,
            },
        });
        const textarea = wrapper.get("textarea");
        await textarea.trigger("focus");
        await textarea.trigger("blur");

        expect(focus).toHaveBeenCalled();
        expect(blur).toHaveBeenCalled();
    });

    it('should have ValidityMode INITIAL when "pending-validity" event is triggered', async () => {
        const wrapper = createWrapper({
            attrs: { id: "elementId" },
        });

        const textarea = wrapper.get("textarea");
        const htmlTextarea = textarea.element as HTMLTextAreaElement;

        htmlTextarea.dispatchEvent(
            new CustomEvent<ValidityEvent>("validity", {
                detail: {
                    target: wrapper.element as ValidatableHTMLElement,
                    elementId: "elementId",
                    isValid: true,
                    validityMode: "ERROR",
                    validationMessage: "Something went wrong.",
                    nativeEvent: "input",
                },
            }),
        );
        await flushPromises();
        wrapper.vm.$forceUpdate();

        expect(wrapper.vm.$data.validityMode).toBe("ERROR");

        htmlTextarea.dispatchEvent(
            new CustomEvent<PendingValidityEvent>("pending-validity"),
        );
        await flushPromises();
        wrapper.vm.$forceUpdate();

        expect(wrapper.vm.$data.validityMode).toBe("INITIAL");
    });
});

it("should warn the user that the maximum string length limit is near", async () => {
    const wrapper = createWrapper({
        props: {
            maxlength: 10,
            softLimit: 5,
            charactersLeftWarning: "Kvar: %charactersLeft%",
        },
    });
    await wrapper.setProps({
        modelValue: "12345",
    });
    expect(wrapper.get('[aria-live="polite"]')).toMatchInlineSnapshot(`
        <label class="label" for="textarea-field" aria-live="polite"><span class="label__description">Kvar: 5</span>
          <!--v-if-->
        </label>
    `);
});

describe("element should be possible to disable with prop disabled", () => {
    it("element should be disabled with prop", async () => {
        const wrapper = mount(FTextareaField, {
            propsData: { disabled: true },
        });
        const element = wrapper.get("textarea").element as HTMLTextAreaElement;
        expect(element.disabled).toBeTruthy();
    });

    it("element should be enabled without prop", async () => {
        const wrapper = mount(FTextareaField, {
            propsData: { disabled: false },
        });
        const element = wrapper.get("textarea").element as HTMLTextAreaElement;
        expect(element.disabled).toBeFalsy();
    });
});

describe("html-validate", () => {
    it.each`
        html
        ${'<f-textarea-field maxlength="10" soft-limit="3">Label</f-textarea-field>'}
    `("$html should be valid", ({ html }) => {
        expect.assertions(1);
        expect(html).toHTMLValidate();
    });

    it.each`
        html
        ${"<f-textarea-field>Label<template #tooltip><div /></template></f-textarea-field>"}
        ${'<f-textarea-field maxlength="100 000" soft-limit="30 000">Label</f-textarea-field>'}
    `("$html should be invalid", ({ html }) => {
        expect.assertions(3);
        let catchedError;

        try {
            expect(html).toHTMLValidate();
        } catch (error) {
            catchedError = error;
        } finally {
            expect(catchedError).toBeDefined();
            expect(catchedError).toMatchSnapshot();
        }
    });
});
