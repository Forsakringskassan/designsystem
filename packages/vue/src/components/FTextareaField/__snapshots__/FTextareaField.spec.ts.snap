// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`html-validate <f-textarea-field maxlength="100 000" soft-limit="30 000">Label</f-textarea-field> should be invalid 1`] = `
[Error: Expected HTML to be valid but had the following errors:

  Attribute "maxlength" has invalid value "100 000" [attribute-allowed-values]
  Attribute "soft-limit" has invalid value "30 000" [attribute-allowed-values]]
`;

exports[`html-validate <f-textarea-field>Label<template #tooltip><div /></template></f-textarea-field> should be invalid 1`] = `
[Error: Expected HTML to be valid but had the following errors:

  <div> element is not permitted as content under slot "tooltip" (<f-textarea-field>) [element-permitted-content]]
`;

exports[`snapshots should match snapshot when validityMode is ERROR and isValid is false 1`] = `
<div
  class="textarea-field textarea-field--error"
>
  <label
    class="label"
    for="textarea-field"
  >
    
    
    Label
    
    
    
    
    <span
      class="label__message label__message--error"
    >
      <f-icon-stub
        class="label__icon--left"
        library="f"
        name="error"
      />
      
      
      
      Something went wrong.
      
      
      
    </span>
  </label>
  <!--v-if-->
  <textarea
    class="textarea-field__textarea textarea-field__resize--none"
    id="textarea-field"
    rows="4"
  />
</div>
`;

exports[`snapshots should match snapshot when validityMode is INITIAL and isValid is false 1`] = `
<div
  class="textarea-field"
>
  <label
    class="label"
    for="textarea-field"
  >
    
    
    Label
    
    
    
    
    <!--v-if-->
  </label>
  <!--v-if-->
  <textarea
    class="textarea-field__textarea textarea-field__resize--none"
    id="textarea-field"
    rows="4"
  />
</div>
`;

exports[`snapshots should match snapshot when validityMode is INITIAL and isValid is true 1`] = `
<div
  class="textarea-field"
>
  <label
    class="label"
    for="textarea-field"
  >
    
    
    Label
    
    
    
    
    <!--v-if-->
  </label>
  <!--v-if-->
  <textarea
    class="textarea-field__textarea textarea-field__resize--none"
    id="textarea-field"
    rows="4"
  />
</div>
`;

exports[`snapshots should match snapshot when validityMode is VALID and isValid is true 1`] = `
<div
  class="textarea-field"
>
  <label
    class="label"
    for="textarea-field"
  >
    
    
    Label
    
    
    
    
    <!--v-if-->
  </label>
  <!--v-if-->
  <textarea
    class="textarea-field__textarea textarea-field__resize--none"
    id="textarea-field"
    rows="4"
  />
</div>
`;

exports[`snapshots should match snapshot with label and textarea 1`] = `
<div
  class="textarea-field"
>
  <label
    class="label"
    for="textarea-field"
  >
    
    
    Label
    
    
    
    
    <!--v-if-->
  </label>
  <!--v-if-->
  <textarea
    class="textarea-field__textarea textarea-field__resize--none"
    id="textarea-field"
    rows="4"
  />
</div>
`;

exports[`snapshots should match snapshot with label, error message and textarea 1`] = `
<div
  class="textarea-field"
>
  <label
    class="label"
    for="textarea-field"
  >
    
    
    Label
    
    
    
    
    <span
      class="label__message label__message--error"
    >
      <f-icon-stub
        class="label__icon--left"
        library="f"
        name="error"
      />
      
      
      ERRROR_MESSAGE
      
      
    </span>
  </label>
  <!--v-if-->
  <textarea
    class="textarea-field__textarea textarea-field__resize--none"
    id="textarea-field"
    rows="4"
  />
</div>
`;

exports[`snapshots should match snapshot with label, tooltip, description, error message and textarea 1`] = `
<div
  class="textarea-field"
>
  <div>
    <div>
      <label
        class="label"
        for="textarea-field"
      >
        
        
        Label
        
        
      </label>
    </div>
    
    
    TOOLTIP
    
    
    <label
      class="label sr-separator"
      for="textarea-field"
    >
      
      
      DESCRIPTION
      
      
      <span
        class="label__message label__message--error"
      >
        <f-icon-stub
          class="label__icon--left"
          library="f"
          name="error"
        />
        
        
        ERROR_MESSAGE
        
        
      </span>
    </label>
  </div>
  <!--v-if-->
  <textarea
    class="textarea-field__textarea textarea-field__resize--none"
    id="textarea-field"
    rows="4"
  />
</div>
`;

exports[`snapshots should match snapshot with resize vertical 1`] = `
<div
  class="textarea-field"
>
  <label
    class="label"
    for="textarea-field"
  >
    
    
    Label
    
    
    
    
    <!--v-if-->
  </label>
  <!--v-if-->
  <textarea
    class="textarea-field__textarea textarea-field__resize--vertical"
    id="textarea-field"
    rows="4"
  />
</div>
`;
